File: src/main/webapp/app/common/secondary/RestStatistics.ts
import { Statistics } from '@/common/domain/Statistics';
import { AxiosResponse } from 'axios';

export interface RestStatistics {
  appliedModules: number;
}

export const toStatistics = (response: AxiosResponse<RestStatistics>): Statistics => {
  const { appliedModules } = response.data;
  return new Statistics(appliedModules);
};

File: src/main/java/tech/jhipster/lite/statistic/infrastructure/primary/RestStatistics.java
package tech.jhipster.lite.statistic.infrastructure.primary;

import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.media.Schema.RequiredMode;
import tech.jhipster.lite.statistic.domain.Statistics;

@Schema(name = "statistics", description = "JHipster lite usage statistics")
class RestStatistics {

  private final long appliedModules;

  private RestStatistics(long appliedModules) {
    this.appliedModules = appliedModules;
  }

  public static RestStatistics from(Statistics statistics) {
    return new RestStatistics(statistics.appliedModules());
  }

  @Schema(description = "Number of module applied on this instance", requiredMode = RequiredMode.REQUIRED)
  public long getAppliedModules() {
    return appliedModules;
  }
}
